src/main/kotlin/com/cognifide/gradle/aem/api/AemConfig.kt

```diff
@@ -421,63 +421,44 @@ class AemConfig(
-    }
-
-    fun localInstance(httpUrl: String, user: String, password: String, type: String) {
-        instance(LocalInstance.create(httpUrl, user, password, type))
-    }
-
-    fun localInstance(httpUrl: String, user: String, password: String, type: String, debugPort: Int) {
-        instance(LocalInstance(httpUrl, user, password, type, debugPort))
+    fun localInstance(httpUrl: String, configurer: LocalInstance.() -> Unit) {
+        instance(LocalInstance.create(httpUrl, configurer))
     }
 
     fun localAuthorInstance() {
-        val httpUrl = propParser.string(Instance.AUTHOR_URL_PROP, Instance.URL_AUTHOR_DEFAULT)
-        instance(LocalInstance.create(httpUrl))
+        localInstance(propParser.string(Instance.AUTHOR_URL_PROP, Instance.URL_AUTHOR_DEFAULT))
     }
 
     fun localPublishInstance() {
-        val httpUrl = propParser.string(Instance.PUBLISH_URL_PROP, Instance.URL_PUBLISH_DEFAULT)
-        instance(LocalInstance.create(httpUrl))
+        localInstance(propParser.string(Instance.PUBLISH_URL_PROP, Instance.URL_PUBLISH_DEFAULT))
     }
 
     fun remoteInstance(httpUrl: String) {
-        instance(RemoteInstance.create(httpUrl, deployEnvironment))
+        instance(RemoteInstance.create(httpUrl, {
+            this.environment = deployEnvironment
+        }))
     }
 
-    fun remoteInstance(httpUrl: String, environment: String) {
-        instance(RemoteInstance.create(httpUrl, environment))
-    }
-
-    fun remoteInstance(httpUrl: String, user: String, password: String) {
-        instance(RemoteInstance.create(httpUrl, user, password, deployEnvironment))
-    }
-
-    fun remoteInstance(httpUrl: String, user: String, password: String, environment: String) {
-        instance(RemoteInstance.create(httpUrl, user, password, environment))
-    }
-
-    fun remoteInstance(httpUrl: String, user: String, password: String, type: String, environment: String) {
-        instance(RemoteInstance(httpUrl, user, password, type, environment))
+    fun remoteInstance(httpUrl: String, configurer: RemoteInstance.() -> Unit) {
+        instance(RemoteInstance.create(httpUrl, {
+            this.environment = deployEnvironment
+            this.apply(configurer)
+        }))
     }
 
     fun remoteAuthorInstance() {
-        val httpUrl = propParser.string(Instance.AUTHOR_URL_PROP, Instance.URL_AUTHOR_DEFAULT)
-        instance(RemoteInstance.create(httpUrl, deployEnvironment))
+        remoteInstance(propParser.string(Instance.AUTHOR_URL_PROP, Instance.URL_AUTHOR_DEFAULT))
     }
 
     fun remotePublishInstance() {
-        val httpUrl = propParser.string(Instance.PUBLISH_URL_PROP, Instance.URL_PUBLISH_DEFAULT)
-        instance(RemoteInstance.create(httpUrl, deployEnvironment))
+        remoteInstance(propParser.string(Instance.PUBLISH_URL_PROP, Instance.URL_PUBLISH_DEFAULT))
     }
 
     private fun instance(instance: Instance) {
+        if (instances.containsKey(instance.name)) {
```

Comment:
added useful validation / it was confusing when some defined instance what not used when deploying